rpc/test/helpers.go

```diff
@@ -23,7 +23,18 @@ import (
 	rpcclient "github.com/tendermint/tendermint/rpc/lib/client"
 )
 
+// Options helps with specifying some parameters for our RPC testing for greater
+// control.
+type Options struct {
```

Comment:
this is a little bit different from what is common in the rest of the codebase 

see https://github.com/tendermint/tendermint/blob/a453628c4e43c67c747d5078704479aa73716182/libs/log/filter.go for example

better to be consistent unless there are strong points for Options struct

Reply:
A good resource to get familiar with options: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

And usage in our codebase for inspiration: https://github.com/tendermint/tendermint/blob/master/p2p/transport.go#L94-L117

Reply:
I originally started with the standard approach of using options to influence the primary object in question (in this case, `*nm.Node`), but the trouble is that that approach assumes all you're only influencing state within that one struct.

For instance, if instead of making options of type `func(*Options)`, I make them of type `func(*nm.Node)` (the way @xla is suggesting), how do I influence state outside of `nm.Node` in `rpctest.GetConfig()` or `rpctest.StartTendermint()` (without introducing global variables and potentially leaking this configuration into other tests)?

There's a bunch of stuff going on inside the `rpc/test/helpers.go` package that isn't encapsulated in a higher-level structure that'd be more amenable to the traditional options approach, I think. Unless you want to refactor to create a higher-level object, something like `RPCTestNode` like so:

```go
type RPCTestNode struct {
    *nm.Node
    suppressStdout bool // This makes the examples work
    recreateConfig bool // This also makes the examples work
}

type Option func(*RPCTestNode)
```

Same goes for `log.filter`: the option function would make sense to be of type `func(*filter)` because all the necessary state to be influenced is contained within `filter`, but that won't allow you to influence any state outside of `filter`.

I thought this alternative approach would work better because it doesn't change the return types: it just adds variadic options to the functions. It also requires less refactoring.

But I can refactor this if consistency is more important?

Reply:
Thanks for all the context, I understand your reasoning. The refactoring you propose would be desirable, but would defo. conflate the scope of this PR. We should capture it in a follow-up issue and focus on the objective of the PR itself for now. Would that make sense?

Reply:
IMHO sounds like this is worth consider doing in a separate PR (?) @melekes @xla

Reply:
@Liamsi see my comment, prior to yours.

Reply:
heh, your comment didn't show while I posted mine. Agree as you can see 👍

Reply:
Added #3565 to address this 👍